internal/db/schema/migrations/oss/postgres/65/01_plugin_host_external_name.up.sql

```diff
@@ -0,0 +1,37 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: MPL-2.0
+
+begin;
+
+  alter table host_plugin_host add column external_name text;
```

Comment:
> I would recommend no leading or trailing white spaces. Another good one would probably be only printable characters.

This is valid for AWS tag names, see example below.

> Do we need to do any escaping to avoid code entered here being used for SQL injection (...)

We use `go-dbw`, which uses `gorm` to write to the database. I'm not entirely sure about the intricacies of `gorm` when it comes to parametrising queries, but I did find [this](https://gorm.io/docs/security.html), alluding to the use of `database/sql` argument placeholders to safeguard against SQL injection. Maybe @tmessi or @jimlambrt can chip in here?

> (...) or impacting how it is displayed in our various UIs/CLIs, etc?

In terms of output, this string is escaped. I set an instance's name to

```
 an instance -%^/&@&#$%^#^&$ \n \t \n name" 
```
(Note: Leading and trailing spaces present in above)

This is the JSON key/val I got for `external_name` after reading it from Boundary:
```json
"external_name": " an instance -%^/&@&#$%^#^&$ \\n \\t \\n name\" "
```

Reply:
> alluding to the use of database/sql argument placeholders to safeguard against SQL injection

Yes, writes done via `db.(Writer).Create` or `db.(Writer).Update` will use gorm which in turn uses parametrized queries. There are places where we do direct insert/updates via `db.(Writer).Exec` but we should also be using parameterized queries (but doesn't look like `Exec` is being used in this case).

Reply:
@talanknight with https://github.com/hashicorp/boundary/pull/3074/commits/6db22d7013339e9a99c5c37a35ea3cdfd7d3bb13 I've added some constraints. I've also tested this with AWS, the most lax of the two. Let me know what your thoughts are on this 😃

Reply:
I'm totally happy with 256-chars and limiting to printable chars only.